### PR TITLE
[Build] Add linker script optimization

### PR DESCRIPTION
**Pull request by Aidyn-A**
This PR adds a linker script optimization based on prioritized symbols that can be extracted from the profiles of popular workloads. The present linker script was generated to target ARM+CUDA and later can be extended if necessary. The reason we target ARM is shown below:

> PyTorch and other applications that access more than 24x 2MB code regions in quick succession can result in performance bottlenecks in the CPU front-end.  The link-time optimization improves executable code locality and improve performance. We recommend turning on the optimization always for PyTorch and other application that behaves similarly.

### CODE REVIEW

**Review by ptrblck — APPROVED** [review]

File: tools/setup_helpers/generate_linker_script.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+def gen_linker_script(
+    filein: str = "cmake/prioritized_text.txt", fout: str = "cmake/linker_script.ld"
+) -> None:
+    with open(filein) as f:
+        prioritized_text = f.readlines()
+        prioritized_text = [
+            line.replace("\n", "") for line in prioritized_text if line != "\n"
+        ]
+    f.close()
```
Comment (malfet) [inline tools/setup_helpers/generate_linker_script.py:12]: You don't need to do `f.close()` as `with open` is already taken care of that

File: tools/setup_helpers/generate_linker_script.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    linker_script_lines = os.popen("ld -verbose").read().split("\n")
```
Comment (malfet) [inline tools/setup_helpers/generate_linker_script.py:14]:
```suggestion
    linker_script_lines = subprocess.check_output(["ld", "-verbose"]).split("\n")
```

**Review by atalman — APPROVED** [review]

**Comment by snadampal** [comment]
Hi @Aidyn-A , much awaited PR!  are there any performance numbers for the models tested with these symbols? how much was the improvement..

**Pull request closed without merging**